boards: shields: esp_8266: Add initial version

nandojve opened the pull request:
Add ESP-8266 modules shield. This shield was tested with SAM4E_XPRO and FRDM-K64F board running samples/net/wifi. Two ESP-8266 were used: the ESP-01 need to be loaded with ESP8266 AT Bin V1.6.2 which is compatible with ESP8266 AT Bin V1.7.x; the ESP-12F was loaded with ESP8266 AT Bin V1.7.3 and requires special partition configuration to works.

nandojve committed:
drivers: wifi: esp: Add esp at version choice name

Add WIFI_ESP_AT_VERSION choice name to allows select version on
*.defconfig files.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: shields: esp_8266: Add initial version

Add ESP-8266 modules shield. The ESP-01 needs to be loaded with
ESP8266 AT Bin V1.6.2 which is compatible with ESP8266 AT Bin
V1.7.x which is the minimal supported version by ESP WIFI driver.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: shields: esp_8266: Add frdm_k64f overlay

Add FRDM-K64F board overlay to enable ESP-8266 WIFI modules.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: shields: esp_8266: Add sam4e_xpro configs

Add SAM4E_XPRO board configurations to enable ESP-8266 WIFI modules.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve commented:
Change from specific configuration to a more generic one since this intents to demonstrate how any ESP-8266 can be used with ESP WIFI offload driver.

stephanosio commented on boards/shields/esp_8266/Kconfig.defconfig:
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2020 Gerson Fernando Budke <nandojve@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+#if SHIELD_ESP_8266
+
+#if NETWORKING
```
I suppose commenting out the `if`s wasn't your intention here?

nandojve replied:
Fixed!

jukkar commented on boards/shields/esp_8266/doc/index.rst:
```diff
@@ -0,0 +1,95 @@
+The ESP-8266 Modules that have 2MB or more is highly recommended upgrade to
+version 1.7.x for better future compatibility.
+
+This tutorial won't explain how firmware must be upgrate. However will present
```
typo s/upgrate/upgraded/
typo s/However will/However, it will/

nandojve replied:
Done!

nandojve committed:
boards: shields: esp_8266: Add disco_l475_iot1 configs

Add DISCO_L475_IOT1 board configurations to enable ESP-8266 WIFI
modules.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve commented:
rebase and added disco_l475_iot1 board configuraton. This board already have es-wifi WIFI driver enabled and shippable fail. Fixed disabling es-wifi and enabling esp-8266. Tested with ESP-8266 on disco_l475_iot1 and now both are working.

nandojve commented on boards/shields/esp_8266/boards/frdm_k64f.overlay:
```diff
@@ -0,0 +1,23 @@
+	status = "disabled";
+};
+
+&arduino_serial {
```
I already tried, &arduino_serial for SAM4E_XPRO got problems. Node was not found for both keep or exclude. Because of that, I keep this way for now.

erwango replied:
@nandojve it seems that indeed, there is no arduino_serial available on this board. Maybe you could just add it ?

nandojve replied:
There isn't an Arduino header available on that board.
![image](https://user-images.githubusercontent.com/7027887/80491783-8d00c300-8939-11ea-8a08-80b85f09e872.png)

erwango replied:
Ha ok. The fact that boards supporting connectors cannot benefit from generic overlay once a board not supporting connectors is added is annoying, but I don't have a better solution right now. Let's move on.

erwango commented on boards/shields/esp_8266/esp_8266.overlay:
```diff
@@ -0,0 +1,7 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* File meant to be empty */
```
Can you add a comment stating why this generic overlay can't be used  ?

nandojve replied:
Done!.

I was thinking about this issue and I was wondering if besides _shield_foo.overlay_ an _arduino.overlay_ could not be useful inside shields directory. I don't know how complex will be evaluate the "feature" arduino on DT and then include _arduino.overlay_. With that, I think, we can solve this kind of issues.

erwango reviewed (changes requested):
Let's comment why generic overlay could not be used.
Otherwise LGTM

erwango commented on boards/shields/esp_8266/esp_8266.overlay:
```diff
@@ -0,0 +1,12 @@
+ * This file can't be used once this shield target all board with
+ * a free UART port. If there is a &arduino_serial declared here,
+ * boards that don't have the compatible shield pins won't work. This
+ * happens because device tree can't handle an inesistent node.
```
inexistent

nandojve replied:
fixed!

erwango reviewed (changes requested):
Last update and we're good to go.

nandojve commented on boards/shields/esp_8266/doc/index.rst:
```diff
@@ -0,0 +1,96 @@
+        - Don't forget to erase Flash before upgrade
+        - Upgrade all binaries at same time.
+        - ESP-8266 bootloader won't send garbage. Try connect at 74880 bps if
+          you module have 26MHz cristal to detect boot fails.
```
Fixed.

jukkar reviewed (approved):
LGTM
